tests/stub/errors/test_errors.py

```diff
@@ -0,0 +1,331 @@
+                self._assert_is_test_error(error, error_data)
+
+    def test_error_classification(self):
+        for as_cause in (False, True):
+            for classification in (
+                "CLIENT_ERROR",
+                "DATABASE_ERROR",
+                "TRANSIENT_ERROR",
+                "SECURITY_ERROR",  # made up classification
+            ):
+                with self.subTest(
+                    as_cause=as_cause,
+                    classification=classification,
+                ):
+                    error_data = self._make_test_error_data(
+                        extra_diag_rec={"_classification": classification},
+                        code=None if as_cause else ...,
+                    )
+                    if as_cause:
+                        error_data = self._make_test_error_data(
+                            cause=error_data,
+                        )
+                    error = self.get_error(error_data)
+                    self._assert_is_test_error(error, error_data)
+
+    def test_filling_default_diagnostic_record(self):
+        for as_cause in (False, True):
+            with self.subTest(as_cause=as_cause):
+                error_data = self._make_test_error_data(
+                    diagnostic_record=None,
+                    code=None if as_cause else ...,
+                )
+                if as_cause:
+                    error_data = self._make_test_error_data(cause=error_data)
+                error = self.get_error(error_data)
+                self._assert_is_test_error(error, error_data)
+
+    def test_filling_default_value_in_diagnostic_record(self):
+        for as_cause in (False, True):
+            for missing_key in (
+                "CURRENT_SCHEMA",
+                "OPERATION",
+                "OPERATION_CODE",
+            ):
+                with self.subTest(as_cause=as_cause, missing_key=missing_key):
+                    error_data = self._make_test_error_data(
+                        del_diag_rec=[missing_key],
+                        code=None if as_cause else ...,
+                    )
+                    if as_cause:
+                        error_data = self._make_test_error_data(
+                            cause=error_data
+                        )
+                    error = self.get_error(error_data)
+                    self._assert_is_test_error(error, error_data)
+
+    def test_keeps_rubbish_in_diagnostic_record(self):
+        use_spacial = self.driver_supports_features(
+            types.Feature.API_TYPE_SPATIAL
+        )
+        for as_cause in (False, True):
+            with self.subTest(as_cause=as_cause):
+                diagnostic_record = {
+                    "foo": "bar",
+                    "_baz": 1.2,
+                    "OPERATION": None,
+                    "CURRENT_SCHEMA": {"uh": "oh!"},
+                    "OPERATION_CODE": False,
+                    "_classification": 42,
```

Comment:
Just to make sure. If this is not string, we stringify it for the `getRawClassification` method. Right?